### PR TITLE
[#IR-315]As a dev, I have limited the auto-refresh to not refresh the filters

### PR DESCRIPTION
**Pull request by Ravi2714**

### CODE REVIEW

**Comment by himanshu-garg-165** [comment]
Please fix the Circle CI errors.

File: src/app/retrospective/retrospective-feedback/retrospective-feedback.component.ts [hunk 1]
```diff
@@ -17,13 +17,15 @@ import {
...
 import { ClickableButtonRendererComponent } from 'app/shared/ag-grid-renderers/clickable-button-renderer/clickable-button-renderer.component';
+import { FilterDataService } from 'app/shared/services/filter-data.service';
 import { RetrospectiveService } from 'app/shared/services/retrospective.service';
 import { UtilsService } from 'app/shared/utils/utils.service';
 import { Subject } from 'rxjs/Subject';
 import 'rxjs/add/operator/takeUntil';
 import { AppConfig } from 'app/app.config';
 import { SuppressKeyboardEventParams } from 'ag-grid/src/ts/entities/colDef';
 import value from '*.json';
+import { Title } from '@angular/platform-browser';
```
[the remark this answers is not in the document]
Comment (Ravi2714) [reply src/app/retrospective/retrospective-feedback/retrospective-feedback.component.ts]: removed

File: src/app/retrospective/retrospective-feedback/retrospective-feedback.component.ts [hunk 2]
```diff
@@ -68,7 +70,8 @@ export class RetrospectiveFeedbackComponent implements OnInit, OnChanges, OnDest
 
     constructor(private retrospectiveService: RetrospectiveService,
         private snackBar: MatSnackBar,
-        private utils: UtilsService) {
+        private utils: UtilsService,
+        private filters: FilterDataService) {
```
[the remark this answers is not in the document]
Comment (Ravi2714) [reply src/app/retrospective/retrospective-feedback/retrospective-feedback.component.ts]: done

File: src/app/shared/services/filter-data.service.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class FilterDataService {
+
+  constructor() { }
```
[the remark this answers is not in the document]
Comment (Ravi2714) [reply src/app/shared/services/filter-data.service.ts]: removed

File: src/app/retrospective/sprint-member-summary/sprint-member-summary.component.ts [hunk 4]
```diff
@@ -216,38 +223,42 @@ export class SprintMemberSummaryComponent implements OnInit, OnChanges, OnDestro
...
                     } else {
                         this.snackBar.open(
                             this.utils.getApiErrorMessage(err) || API_RESPONSE_MESSAGES
                                 .getSprintMemberSummaryError,
-                            '', {duration: SNACKBAR_DURATION});
+                            '', { duration: SNACKBAR_DURATION });
                     }
+                },
+                () => {
+                    this.restoreFilterData();
```
Comment (chahat-arora) [inline src/app/retrospective/sprint-member-summary/sprint-member-summary.component.ts]: Please this fix as discussed.

File: src/app/retrospective/sprint-member-summary/sprint-member-summary.component.ts [hunk 5]
```diff
@@ -198,9 +205,14 @@ export class SprintMemberSummaryComponent implements OnInit, OnChanges, OnDestro
     }
 
     getSprintMemberSummary(isRefresh = false, isAutoRefresh = false) {
+        this.filterService.setFilterData(RETRO_SUMMARY_TYPES.MEMBER, this.gridApi.getFilterModel());
         return this.retrospectiveService.getSprintMemberSummary(this.retrospectiveID, this.sprintID, isAutoRefresh)
             .takeUntil(this.destroy$)
-            .do(
+            .pipe(
```
Comment (himanshu-garg-165) [inline src/app/retrospective/sprint-member-summary/sprint-member-summary.component.ts]: We should make it one-liner.

File: src/app/retrospective/sprint-task-summary/sprint-task-summary.component.ts [hunk 6]
```diff
@@ -194,9 +202,14 @@ export class SprintTaskSummaryComponent implements OnInit, OnChanges, OnDestroy
     }
 
     getSprintTaskSummary(isRefresh = false, isAutoRefresh = false) {
+        this.filterService.setFilterData(RETRO_SUMMARY_TYPES.TASK, this.gridApi.getFilterModel());
         return this.retrospectiveService.getSprintTaskSummary(this.retrospectiveID, this.sprintID, isAutoRefresh)
             .takeUntil(this.destroy$)
-            .do(
+            .pipe(
```
Comment (himanshu-garg-165) [inline src/app/retrospective/sprint-task-summary/sprint-task-summary.component.ts]: Same here.

File: src/app/shared/services/filter-data.service.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    'key-takeaways': [],
+    'things-well-done': [],
+    'additional-things-done': [],
+    'other-highlights': [],
+    'goals-added': [],
+    'goals-accomplished': [],
+    'goals-pending': [],
+    'task-summary': [],
+    'member-summary': [],
+    'task-modal': [],
```
Comment (himanshu-garg-165) [inline src/app/shared/services/filter-data.service.ts]:
```suggestion
    [constants.RETRO_MODAL_TYPES.TASK]: [],
```

File: src/app/shared/services/filter-data.service.ts [hunk 8]
```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class FilterDataService {
+  filterData = {
+    'key-takeaways': [],
```
Comment (himanshu-garg-165) [inline src/app/shared/services/filter-data.service.ts]: As discussed earlier, please use constants here.

File: src/app/retrospective/sprint-member-summary/sprint-member-summary.component.ts (hunk 4 above)
Comment (Ravi2714) [reply src/app/retrospective/sprint-member-summary/sprint-member-summary.component.ts]: done

File: src/app/retrospective/sprint-member-summary/sprint-member-summary.component.ts (hunk 5 above)
Comment (Ravi2714) [reply src/app/retrospective/sprint-member-summary/sprint-member-summary.component.ts]: done

Comment (Ravi2714) [reply src/app/retrospective/sprint-member-summary/sprint-member-summary.component.ts]: done

File: src/app/retrospective/sprint-task-summary/sprint-task-summary.component.ts (hunk 6 above)
Comment (Ravi2714) [reply src/app/retrospective/sprint-task-summary/sprint-task-summary.component.ts]: done

File: src/app/shared/services/filter-data.service.ts (hunk 7 above)
Comment (Ravi2714) [reply src/app/shared/services/filter-data.service.ts]: done

File: src/app/shared/services/filter-data.service.ts (hunk 8 above)
Comment (Ravi2714) [reply src/app/shared/services/filter-data.service.ts]: done

File: src/app/retrospective/retrospective-feedback/retrospective-feedback.component.ts [hunk 9]
```diff
@@ -301,7 +310,8 @@ export class RetrospectiveFeedbackComponent implements OnInit, OnChanges, OnDest
         }
     }
 
-    getTeamMemberOptions(teamMembers) {
+    getTeamMemberOptions() {
```
Comment (chahat-arora) [inline src/app/retrospective/retrospective-feedback/retrospective-feedback.component.ts]: please, Test this properly.

Comment (Ravi2714) [reply src/app/retrospective/retrospective-feedback/retrospective-feedback.component.ts]: fixesd

File: src/app/retrospective/retrospective-feedback/retrospective-feedback.component.ts [hunk 10]
```diff
@@ -128,7 +130,9 @@ export class RetrospectiveFeedbackComponent implements OnInit, OnChanges, OnDest
             stopEditingWhenGridLosesFocus: true,
             suppressDragLeaveHidesColumns: true,
             suppressScrollOnNewData: true,
-            onColumnVisible: (event) => this.gridApi.sizeColumnsToFit()
+            onColumnVisible: (event) => this.gridApi.sizeColumnsToFit(),
+            onFilterChanged: (event) =>{this.filterService.setFilterData(this.feedbackSubType, this.gridApi.getFilterModel());
```
Comment (himanshu-garg-165) [inline src/app/retrospective/retrospective-feedback/retrospective-feedback.component.ts]: Please use this if the statement doesn't exceed the defined line length.
```suggestion
            onFilterChanged: (event) => this.filterService.setFilterData(this.feedbackSubType, this.gridApi.getFilterModel())
```

File: src/app/retrospective/retrospective-feedback/retrospective-feedback.component.ts [hunk 11]
```diff
@@ -568,7 +573,14 @@ export class RetrospectiveFeedbackComponent implements OnInit, OnChanges, OnDest
         if (this.gridApi) {
             this.gridApi.setFilterModel(null);
             this.gridApi.onFilterChanged();
+            this.filterService.setFilterData(this.feedbackSubType, this.gridApi.getFilterModel());
```
Comment (himanshu-garg-165) [inline src/app/retrospective/retrospective-feedback/retrospective-feedback.component.ts]: Do we really need this now?

File: src/app/retrospective/sprint-member-summary/sprint-member-summary.component.ts [hunk 12]
```diff
@@ -147,7 +152,10 @@ export class SprintMemberSummaryComponent implements OnInit, OnChanges, OnDestro
             suppressDragLeaveHidesColumns: true,
             suppressScrollOnNewData: true,
             stopEditingWhenGridLosesFocus: true,
-            onColumnVisible: (event) => this.gridApi.sizeColumnsToFit()
+            onColumnVisible: (event) => this.gridApi.sizeColumnsToFit(),
+            onFilterChanged: (event) =>{
```
Comment (himanshu-garg-165) [inline src/app/retrospective/sprint-member-summary/sprint-member-summary.component.ts]: Same here.

File: src/app/retrospective/sprint-member-summary/sprint-member-summary.component.ts [hunk 13]
```diff
@@ -525,6 +535,12 @@ export class SprintMemberSummaryComponent implements OnInit, OnChanges, OnDestro
         if (this.gridApi) {
             this.gridApi.setFilterModel(null);
             this.gridApi.onFilterChanged();
+            this.filterService.setFilterData(RETRO_SUMMARY_TYPES.MEMBER, this.gridApi.getFilterModel());
```
Comment (himanshu-garg-165) [inline src/app/retrospective/sprint-member-summary/sprint-member-summary.component.ts]: Same here.

File: src/app/retrospective/retrospective-feedback/retrospective-feedback.component.ts (hunk 10 above)
Comment (Ravi2714) [reply src/app/retrospective/retrospective-feedback/retrospective-feedback.component.ts]: done

File: src/app/retrospective/retrospective-feedback/retrospective-feedback.component.ts (hunk 11 above)
Comment (Ravi2714) [reply src/app/retrospective/retrospective-feedback/retrospective-feedback.component.ts]: removed

File: src/app/retrospective/sprint-member-summary/sprint-member-summary.component.ts (hunk 12 above)
Comment (Ravi2714) [reply src/app/retrospective/sprint-member-summary/sprint-member-summary.component.ts]: changed

File: src/app/retrospective/sprint-member-summary/sprint-member-summary.component.ts (hunk 13 above)
Comment (Ravi2714) [reply src/app/retrospective/sprint-member-summary/sprint-member-summary.component.ts]: removed

**Pull request merged**